Testscripts/Linux/channel_change.sh

```diff
@@ -19,229 +13,160 @@
+		while IFS=: read _vmbus_ch _cpu
+		do
+			LogMsg "vmbus: $_vmbus_ch,		cpu id: $_cpu"
+			echo "$_vmbus_ch:$_cpu" >> $OriginalSource
+			if [ $_cpu != 0 ]; then
+				idle_cpus+=($_cpu)
+				# Set 0 to online file, echo 0 > /sys/devices/system/cpu/cpu<number>/online
+				oldState=$(cat /sys/devices/system/cpu/cpu$_cpu/online)
+				echo 0 > /sys/devices/system/cpu/cpu$_cpu/online 2>&1
+				sleep 1
+				newState=$(cat /sys/devices/system/cpu/cpu$_cpu/online)
+				# Verify the cpu id change
+				if [[ $newState = $oldState ]]; then
+					LogMsg "Successfully verified NO cpu state change, because it was used."
+				else
+					LogErr "Failed to verify the cpu state NO change. Expected $oldState, found $newState"
+					FailedCount=$((FailedCount+1))
+				fi
 
-						# Now reset this vmbus's cpu id to 0. The target cpu id is ready to be idle.
-						LogMsg "Found the cpu id, $_cpu_id from the channel $rel_id. Will set 0, the default cpu id, to this cpu"
-						echo 0 > $sysfs_path/channels/$rel_id/cpu
-						sleep 1
-						# validate the change of that cpu id
-						_cpu_id=$(cat $sysfs_path/channels/$rel_id/cpu)
-						if [[ $_cpu_id == "0" ]]; then
-							LogMsg "Successfully changed the cpu id of the channel $rel_id from ${idle_cpus[$cpu_idx]} to 0"
-							cpu_idx=$((cpu_idx+1))
-						else
-							LogErr "Failed to change the cpu id of the channel $rel_id from ${idle_cpus[$cpu_idx]} to 0. Expected 0, but found $_cpu_id"
-							FailedCount=$((FailedCount+1))
-						fi
-					fi
-					read -a line
-				done
-				#LogMsg "Found vmbus channel and its target cpus: ${temp[@]}, ${!temp[@]}"
-			else
-				LogErr "Failed. Supposed to read Sysfs path, but found ${line[@]}"
-				FailedCount=$((FailedCount+1))
+				# Now reset this cpu id to 0.
+				LogMsg "Set the vmbus channel $_vmbus_ch's cpu to the default cpu, 0"
+				echo 0 > $_device/channels/$_vmbus_ch/cpu
+				sleep 1
+				# validate the change of that cpu id
+				_cpu_id=$(cat $_device/channels/$_vmbus_ch/cpu)
+				if [ $_cpu_id = 0 ]; then
+					LogMsg "Successfully set the vmbus channel $_vmbus_ch's cpu to 0"
+					cpu_idx=$((cpu_idx+1))
+				else
+					LogErr "Failed to set the vmbus channel $_vmbus_ch's cpu to 0. Expected 0, but found $_cpu_id"
+					FailedCount=$((FailedCount+1))
+				fi
 			fi
-			idx=$((idx+1))
-		else
-			LogErr "Failed. Supposed to read VMBUS ID line, but found ${line[@]}"
-			FailedCount=$((FailedCount+1))
-		fi
-	done < "$lsvmbus_output_location"
-
-	LogMsg "VMBUS scanning result: ${vmbus_id[@]}"
+			sleep 1
+		done < channel_vp_mapping
+	done
+	LogMsg "The list of target CPU: ${idle_cpus[@]}"
 
 	# #######################################################################################
+	# The previous step sets all channels' cpu to 0, so the rest of non-zero cpu can be offline
 	# Select a CPU number where does not associate to vmbus channels from idle_cpus array
+	LogMsg "Changed all cpu state from online to offline, and vice versa."
```

Comment:
Maybe do s/Changed/Change/ (we're going to change the state)

Reply:
Fixed in 6b1643883a8b1ad8eb082a0e161c107e1bdd9efb